virtualizarr/readers/hdf_filters.py

```diff
@@ -0,0 +1,126 @@
+
+
+class BloscProperties(BaseModel):
+    blocksize: int
+    clevel: int
+    shuffle: int
+    cname: str
+
+    @field_validator("cname", mode="before")
+    def get_cname_from_code(cls, v):
+        blosc_compressor_codes = {
+            value: key
+            for key, value in hdf5plugin._filters.Blosc._Blosc__COMPRESSIONS.items()
+        }
+        return blosc_compressor_codes[v]
+
+
+class ZstdProperties(BaseModel):
+    level: int
+
+
+class CFCodec(TypedDict):
+    target_dtype: np.dtype
+    codec: Codec
+
+
+def _filter_to_codec(
+    filter_id: str, filter_properties: Union[int, None, Tuple] = None
+) -> Codec:
+    id_int = None
+    id_str = None
+    try:
+        id_int = int(filter_id)
+    except ValueError:
+        id_str = filter_id
+    conf = {}
+    if id_str:
+        if id_str in _non_standard_filters.keys():
+            id = _non_standard_filters[id_str]
+        else:
+            id = id_str
+        conf["id"] = id  # type: ignore[assignment]
+        if id == "zlib":
+            conf["level"] = filter_properties  # type: ignore[assignment]
+    if id_int:
+        filter = hdf5plugin.get_filters(id_int)[0]
+        id = filter.filter_name
+        if id in _hdf5plugin_imagecodecs.keys():
+            id = _hdf5plugin_imagecodecs[id]
+        if id == "blosc" and isinstance(filter_properties, tuple):
+            blosc_props = BloscProperties(
+                **{
+                    k: v
+                    for k, v in zip(
+                        BloscProperties.model_fields.keys(), filter_properties[-4:]
+                    )
+                }
+            )
+            conf = blosc_props.model_dump()  # type: ignore[assignment]
+        if id == "zstd" and isinstance(filter_properties, tuple):
+            zstd_props = ZstdProperties(level=filter_properties[0])
+            conf = zstd_props.model_dump()  # type: ignore[assignment]
+
+        conf["id"] = id
+
+    codec = registry.get_codec(conf)
+    return codec
+
+
+def cfcodec_from_dataset(dataset: h5py.Dataset) -> Codec | None:
+    attributes = {attr: dataset.attrs[attr] for attr in dataset.attrs}
+    mapping = {}
+    if "scale_factor" in attributes:
+        try:
+            scale_factor = attributes["scale_factor"][0]
+        except IndexError:
+            scale_factor = attributes["scale_factor"]
+        mapping["scale_factor"] = 1 / scale_factor
+    else:
+        mapping["scale_factor"] = 1
+    if "add_offset" in attributes:
+        try:
+            offset = attributes["add_offset"][0]
+        except IndexError:
+            offset = attributes["add_offset"]
+        mapping["add_offset"] = offset
+    else:
+        mapping["add_offset"] = 0
+    if mapping["scale_factor"] != 1 or mapping["add_offset"] != 0:
+        float_dtype = _choose_float_dtype(dtype=dataset.dtype, mapping=mapping)
+        target_dtype = np.dtype(float_dtype)
+        codec = FixedScaleOffset(
```

Comment:
Are you able to make [this test parametrization](https://github.com/zarr-developers/VirtualiZarr/blob/08ca0cd557d108c1ac500b88fd023142fb3294d7/virtualizarr/tests/test_integration.py#L17) pass with this PR? It's currently xfailed because `open_virtual_dataset` doesn't know how to handle scale factor encoding.

Reply:
I might be misunderstanding, but none of the hdf reader code will be called for `loadable_variables` and this [block](https://github.com/zarr-developers/VirtualiZarr/blob/main/virtualizarr/kerchunk.py#L271) would only be entered for a loaded variable.  Is that correct?